javascript/javascript1/week2/java.js

```diff
@@ -0,0 +1,146 @@
+//Flight booking fullname function
+
+function getFullname(firstname, surname, sex, useFormalName = false) {
+  let prefix = "";
+
+  if (useFormalName) {
+    if (sex === "male") prefix = "Mr";
+    else if (sex === "female") prefix = "Mrs";
+    else console.log("Please enter a valid gender");
+  }
+
+  return `${prefix} ${firstname} ${surname}`;
+}
+fullname1 = getFullname("Benjamin", "Hughes", "male");
+fullname2 = getFullname("Beatrice", "Jackson", "female", true);
+
+console.log(fullname1);
+console.log(fullname2);
+
+//Event application
+
+function getEventWeekday(eventInDays) {
+  const weekdays = [
+    "Sunday",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+  const today = new Date().getDay();
+  const eventDayIndex = (today + eventInDays) % 7;
+
+  return weekdays[eventDayIndex];
+}
+
+console.log(getEventWeekday(3));
+
+//Weather wear
+
+function outfitBasedOnWeather(temp) {
+  if (temp >= 35) {
+    return "shorts and a t-shirt";
+  } else if (temp >= 30) {
+    return "a summer dress ";
+  } else if (temp >= 25) {
+    return "trousers and  shirt";
+  } else if (temp >= 20) {
+    return " light jacket and a  t-shirt";
+  } else if (temp >= 15) {
+    return "Sweater and  pair of jeans";
+  } else if (temp >= 10) {
+    return "Warm pants or jeans";
+  } else {
+    return "Heavy winter coat";
+  }
+}
```

Comment:
Nice one! Can you think of a nested array with all temperatures and strings within. Then a for-loop looping over all conditions with a single `if` ?